lib/elixir/lib/list.ex

```diff
@@ -275,6 +275,26 @@ defmodule List do
     :lists.keydelete(key, position + 1, list)
   end
 
+
+  @doc """
+
+  ## Examples
+
+      iex> List.keytake([a: 1, b: 2], :a, 0)
+      {:value, {:a, 1}, [b: 2]}
+
+      iex> List.keytake([a: 1, b: 2], 2, 1)
+      {:value, {:b, 2}, [a: 1]}
+
+      iex> List.keytake([a: 1, b: 2], :c, 0)
+      false
+
+  """
+  @spec keytake([tuple], any, non_neg_integer) :: {:value, tuple, [tuple]} | false
```

Comment:
Changed to `{tuple, [tuple]} | nil` (which is more alike to original return). How do you feel about adding `keypop/4` (`{any, [tuple]}` and defaults) to have something similar to `Dict.pop/3`?